### PR TITLE
Updated telemetry events

### PR DESCRIPTION
**Pull request by akoutmos**
This PR refactors telemetry events so that they are in their own modules and used in the applicable spots. This cleans up the the consumer and publisher modules and pulls documentation into the modules themselves. This design is largely influenced by Finch (https://github.com/keathley/finch/blob/master/lib/finch/telemetry.ex).

I also updated the fields in the measurement maps to align with the comments in https://github.com/beam-telemetry/telemetry/issues/57.

## Description
Telemetry events have been updated to be inline with Telemetry guidelines.

## Checklist

- [X] I have added unit tests to cover my changes.
- [X] I have improved the code quality of this repo. (refactoring, or reduced number of static analyser issues)
- [X] I have updated the documentation accordingly

### CODE REVIEW

**Comment by akoutmos** [comment]
This one should be good to review now. All changes that I could gather from the various Elixir community projects are in.

**Comment by mkorszun** [comment]
@vorce yes, will take care of the communication. There will be followup PR to update all docs and maybe document migration steps for `2->3`

File: README.md [hunk 1]
```diff
@@ -117,11 +119,6 @@ GenRMQ.Publisher.publish(Publisher, Jason.encode!(%{msg: "msg"}))
 - [Consumer without deadletter configuration][without_deadletter_configuration]
 - [Consumer with quorum queues][with_quorum_queue_type]
 
-### Metrics
```
Comment (mkorszun) [inline README.md:120]: I am not convinced with removing this documentation. We should still treat these modules as internal ones. Users should not interact with them directly meaning that they might not come across their documentation.

**Comment by mkorszun** [comment]
@akoutmos could you elaborate a little about changes in telemetry events? I went quickly through https://github.com/beam-telemetry/telemetry/issues/57 but I do not find it clear.

File: README.md (hunk 1 above)
Comment (akoutmos) [reply README.md]: Good point! I added it back in the README and have the links going to the modules. That work?

**Comment by akoutmos** [comment]
> @akoutmos could you elaborate a little about changes in telemetry events? I went quickly through [beam-telemetry/telemetry#57](https://github.com/beam-telemetry/telemetry/issues/57) but I do not find it clear.

From what I gathered from the various issues and PRs to the other repos, the primary changes were the switch from `time` -> `system_time` in the start event. The standardization of the `exception` event when an exception occurs and a stacktrace is possibly available. Also the fact that the `stop` event can have an optional `error` field to denote that something went wrong...but not necessarily an exception with a stacktrace.

File: README.md (hunk 1 above)
Comment (mkorszun) [reply README.md]: @akoutmos yes, sounds good 👍

**Comment by akoutmos** [comment]
This one should be good to merge now I imagine ? The CI error is because 1 of the jobs failed to publish to coveralls and the link check will fail until the docs make their way to master.

**Comment by mkorszun** [comment]
@akoutmos great, will merge it soon. Great job 👍

**Pull request merged**